lib/elixir/src/elixir_parser.yrl

```diff
@@ -813,3 +817,8 @@ throw_no_parens_container_strict(Node) ->
     "adding parentheses:\n\n"
     "    [one, two(three, four), five]\n\n"
     "Elixir cannot compile otherwise. Syntax error before: ", "','").
+
+throw_empty_stab_clause(StabOpToken) ->
+  throw(meta_from_token(StabOpToken),
+        "no expression after the ->. Syntax error before: ",
```

Comment:
What about "an expression is required on the right side of ->. Syntax error before: " ?